### ISSUE TITLE
🐛 imports failing due to cryptic `_computedKey` error (SWC-related)

### ISSUE THREAD

**Issue by nberlette** [nberlette/deno_fmt#2]
Imports of the module are currently failing in Deno Deploy, throwing a cryptic error about `_computedKey` being undefined on line 19 of `options.ts`.

However, the identifier `"_computedKey"` is not used once acrosss this entire project. This same error has been appearing for users of 1.40+ recently when using the new Decorators syntax, and is caused by the SWC compiler's buggy implementation of the Decorators Proposal.... but why it has decided to show up in this project, so far I cannot figure out. 

I think it might be something to do with the awful code I wrote in options.ts, so I'm going to just nuke that file and rewrite it from the ground up.

**Comment by alexgshaw** [comment]
Running into this issue with my Supabase deno edge function.

Deno: 1.40.3

```
runtime has escaped from the event loop unexpectedly: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
failed to send request to user worker: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
error in user worker connection: connection closed before message completed
user worker failed to respond: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
TypeError: request body receiver not connected (channel closed)
    at Object.write (ext:deno_web/06_streams.js:1155:20)
    at Module.invokeCallbackFunction (ext:deno_webidl/00_webidl.js:944:16)
    at WritableStreamDefaultController.writeAlgorithm (ext:deno_web/06_streams.js:3915:14)
    at writableStreamDefaultControllerProcessWrite (ext:deno_web/06_streams.js:4485:55)
    at writableStreamDefaultControllerAdvanceQueueIfNeeded (ext:deno_web/06_streams.js:4388:5)
    at writableStreamDefaultControllerWrite (ext:deno_web/06_streams.js:4532:3)
    at writableStreamDefaultWriterWrite (ext:deno_web/06_streams.js:4679:3)
    at Object.chunkSteps (ext:deno_web/06_streams.js:2743:17)
    at readableStreamFulfillReadRequest (ext:deno_web/06_streams.js:2585:17)
    at readableByteStreamControllerCommitPullIntoDescriptor (ext:deno_web/06_streams.js:2276:5)
```

**Comment by nberlette** [comment]
@alexgshaw I forgot to close this issue as fixed, but this was taken care of ~1.41.0 (IIRC).

I highly recommend upgrading your Deno version to the latest release, if you haven't already. They just rolled out [1.44.0](https://deno.com/blog/v1.44) and the improvements made in just 4 minor versions have been **very** significant.

**Comment by nberlette** [comment]
@alexgshaw 

Sorry, I'm up a bit past my bed time 😅 I missed that part where you said supabase edge functions, so disregard my comments about upgrading. I haven't played with edge functions recently, but it's a bit disappointing to hear they're still running v1.40.x nearly 5 months after it was released. 

Even more so considering that version range was probably the most buggy set of releases I've ever used of Deno. It was (essentially) all caused by upstream bugs with SWC that Deno depends on, specifically with the rolling out of the support for Stage 3 Decorators. 

> To put the bugginess in perspective, that `_computedKey` error rendered almost all my personal projects (and many of my clients', too) on Deno Deploy totally useless. For a whole weekend. It wasn't fun.

But that's all been fixed for a while now... I think it was in February when they patched all that up. I'll see if I can reach out to the Supabase team and push them to upgrade to a more recent stable release; I recommend you try doing the same.

I'll leave this open for a bit longer in case you feel the need to reply to any of my late night ramblings. Best of luck to you, Alex!

- Nick
